lib/vsc/filesystem/posix.py

```diff
@@ -72,31 +72,15 @@ def __init__(self):
 
     def _execute(self, cmd, changes=False):
         """Run command cmd, return exitcode,output"""
-        ec = None
-        res = None
-
-        shell = False
-        if isinstance(cmd, (tuple, list,)):
-            # cmd in non-shell
-            shell = False
-        elif isinstance(cmd, str):
-            shell = True
-        else:
-            self.log.error("_execute unsupported type %s of cmd %s" % (type(cmd), cmd))
-            return ec, res
-
         if changes and self.dry_run:
-            self.log.info("Dry run: not really executing cmd %s" % (cmd))
+            self.log.info("Dry run: not really executing cmd %s", cmd)
             return 0, ""
 
         # Executes and captures the output of a succesful run.
-        try:
-            out = subprocess.check_output(cmd, shell=shell)
```

Comment:
i don't know much about the inner workings of this, but `run` doesn't handle large output well, better to use `run_asyncloop` here to be on the safe side